### PR TITLE
[frontend-api] added promoted products

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| With frontend API is now possible to get promoted products
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Review by s3tezsky — APPROVED** [review]

File: packages/frontend-api/apiary.apib [hunk 1]
```diff
@@ -2573,3 +2573,339 @@ Returns brand filtered using UUID
...
+                name
+            }
+        }
+    }
+
+- Response 200 (application/json; charset=utf-8)
+
+    {
+        "data": {
+            "topProducts": [
```
Comment (sslt) [inline packages/frontend-api/apiary.apib:2629]: please, rename to something more fitting, say "promotedProducts"
also, I would like to see similar english H1 title on admin page, so it is obvious, whats what.

Comment (grossmannmartin) [reply packages/frontend-api/apiary.apib]: fixed, thanks!

File: project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        ';
+
+        $graphQlType = 'product';
+        $responseProductDetail = $this->getResponseContentForQuery($queryProductDetail);
+
+        $this->assertResponseContainsArrayOfDataForGraphQlType($responseProductDetail, $graphQlType);
+        $responseDataProductDetail = $this->getResponseDataForGraphQlType($responseProductDetail, $graphQlType);
+
+        self::assertEquals($responseDataPromotedProducts[0], $responseDataProductDetail);
```
Comment (s3tezsky) [inline project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php:121]: This can lead to undefined offset error. The failed test sounds more reasonable to me in that case.

File: project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        ';
+
+        $graphQlType = 'promotedProducts';
+        $responsePromotedProducts = $this->getResponseContentForQuery($queryPromotedProducts);
+
+        $this->assertResponseContainsArrayOfDataForGraphQlType($responsePromotedProducts, $graphQlType);
+        $responseDataPromotedProducts = $this->getResponseDataForGraphQlType($responsePromotedProducts, $graphQlType);
+
+        $productUuid = $responseDataPromotedProducts[0]['uuid'];
```
Comment (s3tezsky) [inline project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php:79]: This can lead to undefined offset errors. The failed test sounds more reasonable to me in that case.

File: upgrade/UPGRADE-v9.1.0-dev.md [hunk 4]
```diff
@@ -144,3 +144,8 @@ There you can find links to upgrade notes for other versions too.
 
 - add additional data to Product frontend API type ([#2057](https://github.com/shopsys/shopsys/pull/2057))
     - see #project-base-diff to update your project
+
+- add promoted products in frontend API ([#2066](https://github.com/shopsys/shopsys/pull/2066))
+    - translation in the administration was changed from "Main page products" to "Promoted products"
+        - translation ID will be changed accordingly in the next major
```
Comment (s3tezsky) [inline upgrade/UPGRADE-v9.1.0-dev.md:150]: I am not really sure how we handle it, there is no deprecation sign. Should it be there?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 5]
```diff
@@ -1340,7 +1340,7 @@ msgid "Main image"
 msgstr "Hlavní obrázek"
 
 msgid "Main page products"
-msgstr "Zboží na titulní stránce"
+msgstr "Propagované produkty"
```
Comment (s3tezsky) [inline packages/framework/src/Resources/translations/messages.cs.po:1343]: Is this change really needed in this PR?

File: upgrade/UPGRADE-v9.1.0-dev.md (hunk 4 above)
Comment (grossmannmartin) [reply upgrade/UPGRADE-v9.1.0-dev.md:150]: Part of removing deprecations before a major release is checking for changed translations in messages and update msgids accordingly, so all English translations have empty msgstr.

File: project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php (hunk 3 above)
Comment (grossmannmartin) [reply project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php:82]: thanks, fixed in 6b185d6fa

File: project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php (hunk 2 above)
Comment (grossmannmartin) [reply project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php:125]: thanks, fixed in 6b185d6fa

File: upgrade/UPGRADE-v9.1.0-dev.md (hunk 4 above)
Comment (s3tezsky) [reply upgrade/UPGRADE-v9.1.0-dev.md:150]: Ok, thanks!

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 5 above)
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po:1343]: see https://github.com/shopsys/shopsys/pull/2066#discussion_r503730973

But when you mentioning it, it could be probably separated

**Review by s3tezsky — APPROVED** [review]

**Review by sslt — APPROVED** [review]
